src/Microsoft.AspNet.FileProviders.Common/Implementation/EnumerableDirectoryContents.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.FileProviders
+{
+    public class EnumerableDirectoryContents : IDirectoryContents
```

Comment:
I'm still a bit confused... if these shared files are shared into each assembly, won't these *public* types now be duplicated in several assemblies? Should every type in this package be internal?